qa/support/logstash-integration-failure_injector/lib/logstash/outputs/failure_injector.rb

```diff
@@ -0,0 +1,72 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# encoding: utf-8
+
+require 'logstash/inputs/base'
+require 'logstash/namespace'
+
+require_relative '../utils/failure_injector_util'
+
+class LogStash::Outputs::FailureInjector < LogStash::Outputs::Base
+
+  config_name "failure_injector"
+
+  # one or any of [register, receive, close]
```

Comment:
The intent of `degrate_at` and also for `crash_at` should state that these are the various phases of a plugin lifecycle

Reply:
Maybe a description that cite that correspond to the phases of a plugin would be helpful, I think.

Reply:
Yup, sorry missed it and addressed with [this commit](https://github.com/elastic/logstash/pull/16466/commits/4dfd471817ac2ea1d270c159b4ec6b6f5e0674dd).